### PR TITLE
cointop 1.0.5 (new formula)

### PR DESCRIPTION
**Pull request by miguelmota**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

add [cointop](https://github.com/miguelmota/cointop) 1.0.4

### CODE REVIEW

File: Formula/cointop.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Cointop < Formula
+  desc "Interactive terminal based UI application for tracking cryptocurrencies"
+  homepage "https://cointop.sh"
+  url "https://github.com/miguelmota/cointop/archive/1.0.0.tar.gz"
+  sha256 "8ff6988cd18b35dbf85436add19135a587e03702b43744f563f137bb067f6e04"
+  revision 1
```
Comment (commitay) [inline Formula/cointop.rb]: Remove revision

File: Formula/cointop.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  head "https://github.com/miguelmota/cointop.git"
```
Comment (commitay) [inline Formula/cointop.rb]: Remove head

File: Formula/cointop.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
```
Comment (commitay) [inline Formula/cointop.rb]:
```ruby
  def install
    ENV["GOPATH"] = buildpath
    (buildpath/"src/github.com/miguelmota/cointop").install buildpath.children
    
    cd "src/github.com/miguelmota/cointop" do
      system "go", "build", "-o", bin/"cointop"
      prefix.install_metafiles
    end
  end
```

**Review by commitay — CHANGES_REQUESTED** [review]

**Comment by miguelmota** [comment]
@commitay suggestions implemented

File: Formula/cointop.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Cointop < Formula
+  desc "Interactive terminal based UI application for tracking cryptocurrencies"
+  homepage "https://cointop.sh"
+  url "https://github.com/miguelmota/cointop/archive/1.0.0.tar.gz"
+  sha256 "8ff6988cd18b35dbf85436add19135a587e03702b43744f563f137bb067f6e04"
+  depends_on "go" => :build
```
Comment (commitay) [inline Formula/cointop.rb]: Newline between sha256 and depends_on

File: Formula/cointop.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/miguelmota/cointop").install buildpath.children
+    cd "src/github.com/miguelmota/cointop" do
+      system "go", "build", "-o", "#{bin}/cointop"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    system "true"
```
Comment (commitay) [inline Formula/cointop.rb]: The test needs to exercise and verifiy the functionality of the software

**Comment by miguelmota** [comment]
@commitay added test

**Comment by miguelmota** [comment]
The following properly works

```ruby
  test do
    system "TERM=screen-256color #{bin}/cointop -test"
  end
```

However, the homebrew "suggested" way from running audit fails:

```ruby
  test do
    system "TERM=screen-256color", "#{bin}/cointop", "-test"
  end
```

```
Testing cointop
==> TERM=screen-256color /usr/local/Cellar/cointop/1.0.1/bin/cointop -test
Last 15 lines from /Users/mota/Library/Logs/Homebrew/cointop/test.01.TERM=screen-256color:
2018-04-29 11:37:25 -0700

TERM=screen-256color
/usr/local/Cellar/cointop/1.0.1/bin/cointop
-test

Error: cointop: failed
Failed executing: TERM=screen-256color /usr/local/Cellar/cointop/1.0.1/bin/cointop -test
/usr/local/Homebrew/Library/Homebrew/formula.rb:1820:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1758:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1758:in `system'
/usr/local/opt/cointop/.brew/cointop.rb:19:in `block in <class:Cointop>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1664:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:556:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1663:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:838:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1662:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `block in run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:12:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1656:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:28:in `block in <main>'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:27:in `<main>'
```

**Comment by miguelmota** [comment]
ping

**Comment by miguelmota** [comment]
@woodruffw ah I see

thanks!

File: Formula/cointop.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/miguelmota/cointop").install buildpath.children
+    cd "src/github.com/miguelmota/cointop" do
+      system "go", "build", "-o", "#{bin}/cointop"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    pipe_output("#{bin}/cointop -test", nil, 0)
```
Comment (woodruffw) [inline Formula/cointop.rb]: Is this right? `pipe_output` is normally used to feed input into a command or test a command's output, but you're doing neither here.

Comment (miguelmota) [reply Formula/cointop.rb]: @woodruffw good point

reverted to using `system`. if `cointop -test` returns exit code 0 then it's successful

**Comment by woodruffw** [comment]
Shoo, bot.

This LGTM -- would you mind merging this @commitay (unless you have additional feedback)? I won't be in front of a Mac for a while :slightly_smiling_face:

**Comment by DomT4** [comment]
There's a 1.0.5 release available if OP wants to grab that, but if not this seems reasonable to me as well.

**Comment by commitay** [comment]
Thanks for the 🆕 formula @miguelmota!

**Pull request closed without merging**